### ISSUE TITLE
PAM Auth using SSSD not creating home directories upon login for new users

### ISSUE THREAD

**Issue by whositwhatnow** [jupyterhub/jupyterhub#3106]
### Bug description
New users home directories are not being created on user login

#### Expected behaviour
New users login and home directories get automatically created once the spawning occurs

#### Actual behaviour
Users authentication works but home directories are not created.  Get error below.  If the home directory is created prior to login then it works.

Jul  3 09:37:22 jupyterhub jupyterhub: [I 2020-07-03 09:37:22.647 JupyterHub base:707] User logged in: newuser
Jul  3 09:37:22 jupyterhub jupyterhub: [I 2020-07-03 09:37:22.652 JupyterHub log:174] 302 POST /hub/login?next= -> /hub/spawn (newuser@1.1.1.1) 363.10ms
Jul  3 09:37:22 jupyterhub jupyterhub: [I 2020-07-03 09:37:22.734 JupyterHub systemdspawner:217] user:newuser Unit jupyter-newuser-singleuser in a failed state. Resetting state.
Jul  3 09:37:22 jupyterhub jupyterhub: Running as unit jupyter-newuser-singleuser.service.
Jul  3 09:37:22 jupyterhub systemd: Started /bin/bash -c cd /home/newuser && exec jupyterhub-singleuser --port=42734.
Jul  3 09:37:22 jupyterhub bash: /bin/bash: line 0: cd: /home/newuser: No such file or directory
Jul  3 09:37:22 jupyterhub systemd: jupyter-newuser-singleuser.service: main process exited, code=exited, status=1/FAILURE
Jul  3 09:37:22 jupyterhub systemd: Unit jupyter-newuser-singleuser.service entered failed state.
Jul  3 09:37:22 jupyterhub systemd: jupyter-newuser-singleuser.service failed.

- OS: CentOS Linux release 7.8.2003 (Core)
- Jupyterhub Version: 1.1.0
- SystemdSpawner

**Comment by danfulton** [comment]
I've been trying to modify a PAM auth scheme and also had trouble with JupyterHub.  Have you tried adding the `pam_mkhomedir.so` to `\etc\pam.d.\common-session` ?

**Comment by whositwhatnow** [comment]
Tried a bunch of things related to PAM itself.
When users ssh into the server it works new home is created, so PAM is working.  Its only when new users attempt to login to our Juptyerhub instance.  Auth works but it doesn't spawn because home directories are not being created.

Instead I got around it by using a prehook and a batch script.  Here is the example: https://github.com/jupyterhub/jupyterhub/tree/master/examples/bootstrap-script
